From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Thu, 31 Jul 2025 21:45:17 +0200
Subject: fix(kernel-modules): add Cadence USB driver to base

Installed Debian Trixie d-i rc2 (dvd-1) on USB storage on Pine64 Star64
is not bootable due to missing modules `cdns3_starfive` and `cdns3` from
initramfs.

Note the above USB support on Pine64 Star64 is directly part of StarFive
JH7110 CPU and differs from StarFive VisionFive2 which uses an external
controller attached to the PCIe bus.

Bug-Debian: https://bugs.debian.org/1108924
Forwarded: https://github.com/dracut-ng/dracut-ng/pull/1493
---
 modules.d/70kernel-modules/module-setup.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules.d/70kernel-modules/module-setup.sh b/modules.d/70kernel-modules/module-setup.sh
index 52d850c..9c98d3a 100755
--- a/modules.d/70kernel-modules/module-setup.sh
+++ b/modules.d/70kernel-modules/module-setup.sh
@@ -93,6 +93,7 @@ installkernel() {
                 "=drivers/soc" \
                 "=drivers/spi" \
                 "=drivers/spmi" \
+                "=drivers/usb/cdns3" \
                 "=drivers/usb/chipidea" \
                 "=drivers/usb/dwc2" \
                 "=drivers/usb/dwc3" \
